items:
- uid: Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException
  commentId: T:Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException
  id: RuntimeBinderInternalCompilerException
  parent: Microsoft.CSharp.RuntimeBinder
  children:
  - Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException.#ctor
  - Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException.#ctor(System.String)
  - Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  - vb
  name: RuntimeBinderInternalCompilerException
  fullName: Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException
  type: Class
  source:
    remote:
      path: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinderInternalCompilerException.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: RuntimeBinderInternalCompilerException
    path: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinderInternalCompilerException.cs
    startLine: 13
  assemblies:
  - Microsoft.CSharp
  namespace: Microsoft.CSharp.RuntimeBinder
  summary: "\n<p>Represents an error that occurs when a dynamic bind in the C# runtime binder is processed.</p>\n"
  remarks: "\n<p>Exceptions of this kind differ from <xref href=\"Microsoft.CSharp.RuntimeBinder.RuntimeBinderException\" data-throw-if-not-resolved=\"false\"></xref> in that <xref href=\"Microsoft.CSharp.RuntimeBinder.RuntimeBinderException\" data-throw-if-not-resolved=\"false\"></xref> represents a failure to bind in the sense of a usual compiler error, whereas <xref href=\"Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException\" data-throw-if-not-resolved=\"false\"></xref> represents a malfunctioning of the runtime binder itself.</p>\n"
  example:
  - 
  syntax:
    content: 'public class RuntimeBinderInternalCompilerException : System.Exception'
    content.vb: >-
      Public Class RuntimeBinderInternalCompilerException
          Inherits System.Exception
  inheritance:
  - System.Object
  modifiers.csharp:
  - public
  - class
  modifiers.vb:
  - Public
  - Class
- uid: Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException.#ctor
  commentId: M:Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException.#ctor
  id: '#ctor'
  parent: Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException
  langs:
  - csharp
  - vb
  name: RuntimeBinderInternalCompilerException()
  fullName: Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException.RuntimeBinderInternalCompilerException()
  type: Constructor
  source:
    remote:
      path: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinderInternalCompilerException.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinderInternalCompilerException.cs
    startLine: 18
  assemblies:
  - Microsoft.CSharp
  namespace: Microsoft.CSharp.RuntimeBinder
  summary: "\n<p>Initializes a new instance of the <xref href=\"Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException\" data-throw-if-not-resolved=\"false\"></xref> class with a system-supplied message that describes the error.</p>\n"
  remarks: "\n<p>This constructor initializes the <xref href=\"System.Exception.Message\" data-throw-if-not-resolved=\"false\"></xref> property of the new instance to a system-supplied message that describes the error, such as \"DefaultMessageDisplayedByParameterlessCtorWriterMustSupply\" This message takes into account the current system culture.</p>\n<p>The following table shows the initial property values for an instance of <xref href=\"System.InsufficientMemoryException\" data-throw-if-not-resolved=\"false\"></xref>.</p>\n<p>Property </p>\n<p>Value </p>\n<table><tbody><tr><td>\n<p>\n<xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>\n<strong>null</strong>.</p>\n</td></tr><tr><td>\n<p>\n<xref href=\"System.Exception.Message\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>The localized error message string. </p>\n</td></tr></tbody></table>\n"
  example:
  - 
  syntax:
    content: public RuntimeBinderInternalCompilerException()
    content.vb: Public Sub New
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
- uid: Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException.#ctor(System.String)
  commentId: M:Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException
  langs:
  - csharp
  - vb
  name: RuntimeBinderInternalCompilerException(String)
  fullName: Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException.RuntimeBinderInternalCompilerException(System.String)
  type: Constructor
  source:
    remote:
      path: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinderInternalCompilerException.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/RuntimeBinderInternalCompilerException.cs
    startLine: 27
  assemblies:
  - Microsoft.CSharp
  namespace: Microsoft.CSharp.RuntimeBinder
  summary: "\n<p>Initializes a new instance of the <xref href=\"Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException\" data-throw-if-not-resolved=\"false\"></xref> class with a specified message that describes the error.</p>\n"
  remarks: "\n<p>The following table shows the initial property values for an instance of <xref href=\"Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException\" data-throw-if-not-resolved=\"false\"></xref>.</p>\n<p>Property </p>\n<p>Value </p>\n<table><tbody><tr><td>\n<p>\n<xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>\n<strong>null</strong>.</p>\n</td></tr><tr><td>\n<p>\n<xref href=\"System.Exception.Message\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>The error message string specified in <em>message</em>. </p>\n</td></tr></tbody></table>\n"
  example:
  - 
  syntax:
    content: public RuntimeBinderInternalCompilerException(string message)
    content.vb: Public Sub New(message As String)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>The message that describes the exception. The caller of this constructor is required to ensure that this string has been localized for the current system culture.</p>\n"
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
- uid: Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException.#ctor(System.String,System.Exception)
  commentId: M:Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException
  langs:
  - csharp
  - vb
  name: RuntimeBinderInternalCompilerException(String, System.Exception)
  fullName: Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException.RuntimeBinderInternalCompilerException(System.String, System.Exception)
  type: Constructor
  source:
    remote:
      path: src/Microsoft.CSharp/ref/Microsoft.CSharp.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/Microsoft.CSharp/ref/Microsoft.CSharp.cs
    startLine: 68
  assemblies:
  - Microsoft.CSharp
  namespace: Microsoft.CSharp.RuntimeBinder
  summary: "\n<p>Initializes a new instance of the <xref href=\"Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException\" data-throw-if-not-resolved=\"false\"></xref> class that has a specified error message and a reference to the inner exception that is the cause of this exception.</p>\n"
  remarks: "\n<p>An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property. The <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property returns the same value that is passed into the constructor, or <strong>null</strong> if the <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property does not supply the inner exception value to the constructor.</p>\n<p>The following table shows the initial property values for an instance of <xref href=\"Microsoft.CSharp.RuntimeBinder.RuntimeBinderInternalCompilerException\" data-throw-if-not-resolved=\"false\"></xref>.</p>\n<p>Property </p>\n<p>Value </p>\n<table><tbody><tr><td>\n<p>\n<xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>\n<strong>null</strong>. </p>\n</td></tr><tr><td>\n<p>\n<xref href=\"System.Exception.Message\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>The error message string specified in <em>message</em>. </p>\n</td></tr></tbody></table>\n"
  example:
  - 
  syntax:
    content: public RuntimeBinderInternalCompilerException(string message, System.Exception innerException)
    content.vb: Public Sub New(message As String, innerException As System.Exception)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>The error message that explains the reason for the exception.</p>\n"
    - id: innerException
      type: System.Exception
      description: "\n<p>The exception that is the cause of the current exception, or a null reference if no inner exception is specified.</p>\n"
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
references:
- uid: Microsoft.CSharp.RuntimeBinder
  isExternal: false
  name: Microsoft.CSharp.RuntimeBinder
  fullName: Microsoft.CSharp.RuntimeBinder
- uid: System.Object
  parent: System
  isExternal: false
  name: Object
  fullName: System.Object
- uid: System
  isExternal: false
  name: System
  fullName: System
- uid: System.String
  parent: System
  isExternal: false
  name: String
  fullName: System.String
- uid: System.Exception
  parent: System
  isExternal: false
  name: Exception
  fullName: System.Exception
